src/createTokenBridge.integration.test.ts

```diff
@@ -0,0 +1,308 @@
+  expect(tokenBridgeContracts.parentChainContracts.wethGateway).not.toEqual(zeroAddress);
+  expect(tokenBridgeContracts.parentChainContracts.weth).not.toEqual(zeroAddress);
+  expect(tokenBridgeContracts.parentChainContracts.multicall).not.toEqual(zeroAddress);
+
+  // child chain contracts
+  expect(Object.keys(tokenBridgeContracts.childChainContracts)).toHaveLength(9);
+  expect(tokenBridgeContracts.childChainContracts.router).not.toEqual(zeroAddress);
+  expect(tokenBridgeContracts.childChainContracts.standardGateway).not.toEqual(zeroAddress);
+  expect(tokenBridgeContracts.childChainContracts.customGateway).not.toEqual(zeroAddress);
+  expect(tokenBridgeContracts.childChainContracts.wethGateway).not.toEqual(zeroAddress);
+  expect(tokenBridgeContracts.childChainContracts.weth).not.toEqual(zeroAddress);
+  expect(tokenBridgeContracts.childChainContracts.proxyAdmin).not.toEqual(zeroAddress);
+  expect(tokenBridgeContracts.childChainContracts.beaconProxyFactory).not.toEqual(zeroAddress);
+  expect(tokenBridgeContracts.childChainContracts.upgradeExecutor).not.toEqual(zeroAddress);
+  expect(tokenBridgeContracts.childChainContracts.multicall).not.toEqual(zeroAddress);
+});
+
+it(`successfully deploys token bridge contracts with a custom fee token through token bridge creator`, async () => {
+  const testnodeInformation = getInformationFromTestnode();
+
+  // deploy a fresh token bridge creator, because it is only possible to deploy one token bridge per rollup per token bridge creator
+  const tokenBridgeCreator = await deployTokenBridgeCreator({
+    publicClient: nitroTestnodeL2Client,
+  });
+
+  // -----------------------------
+  // 1. fund l3deployer account
+  const fundTxRequestRaw = await nitroTestnodeL2Client.prepareTransactionRequest({
+    chain: nitroTestnodeL2Client.chain,
+    to: testnodeInformation.l3NativeToken,
+    data: encodeFunctionData({
+      abi: erc20.abi,
+      functionName: 'transfer',
+      args: [l3RollupOwner.address, parseEther('500')],
+    }),
+    value: BigInt(0),
+    account: l3TokenBridgeDeployer,
+  });
+
+  // sign and send the transaction
+  const fundTxRequest = { ...fundTxRequestRaw, chainId: nitroTestnodeL2Client.chain.id };
+  const fundTxHash = await nitroTestnodeL2Client.sendRawTransaction({
+    serializedTransaction: await l3TokenBridgeDeployer.signTransaction(fundTxRequest),
+  });
+
+  // get the transaction receipt after waiting for the transaction to complete
+  const fundTxReceipt = await nitroTestnodeL2Client.waitForTransactionReceipt({
+    hash: fundTxHash,
+  });
+  expect(fundTxReceipt.status).toEqual('success');
+
+  // -----------------------------
+  // 2. check the new token bridge creator allowance
+  // NOTE: We will estimate the token bridge creation through the canonical TokenBridgeCreator
+  //       and we will later change the destination address to the new deployed TokenBridgeCreator.
+  //       Thus, we need to give allowance to both TokenBridgeCreators
```

Comment:
This is what I mentioned earlier about the tests being a bit messy. The ideal solution might not be feasible, but we could try to find a way of not having to modfify the requests received from the functions that we are testing.